teps/0015-pause-pipeline.md

```diff
@@ -0,0 +1,122 @@
+
+`PipelineRuns` can be created paused so that they are not started at all or the
+pause setting can be used to prevent further `TaskRuns` from being created. As
+it is not possible to "pause" a Kubernetes pod, the pause setting would not
+affect `TaskRuns` that have already been created.
+
+## Motivation
+
+The primary motivation for this is to support operators who need to control the
+start up of `PipelineRuns` in clusters that are under heavy load. By allowing
+`PipelineRuns` to be created without actually starting them, one could implement
+a custom controller to implement any `PipelineRun` scheduling policy that is
+transparent to the `PipelineRun` controller and `PipelineRun` creator (e.g.,
+Tekton triggers).
+
+With this feature, platforms can control when `PipelineRuns` are started without
+introducing a bespoke system for storing `PipelineRuns` - potentially adding new
+failure scenarios or UI complexity.
+
+### Goals
+
+* Provide a pause mechanism for `PipelineRuns` that operators can use
+  to control whether or not a `PipelineRun` is running (e.g., "creating new
+  `TaskRuns`").
+
+### Non-Goals
+
+* Support pausing of a running `TaskRun` as that would requiring pausing the
+  Kubernetes pods themselves.
+
+## Requirements
+
+* Provide a mechanism to operators that can be used to pause a `PipelineRun`.
+* `PipelineRuns` that are created with the pause setting enabled are not started
+  until the setting is toggled off.
+* `PipelineRuns` that have the pause setting enabled after they are started will
+  not create any new `TaskRuns` until the setting is toggled off.
+
+## Proposal
+
+The `PipelineRunSpec` supports a field called `Status` of the type
+`PipelineRunSpecStatus`. Currently the field only supports a single setting,
+`PipelineRunCancelled`. This field will be expanded to support an additional
+setting, `PipelineRunPaused`.
+
+When the `Status` field is set to `PipelineRunPaused`, the Tekton reconciler
+will set the `Succeeded` `Condition` `Reason` in the `PipelineRun` `Status` to
+`PipelineRunPaused`:
+
+```
+status:
+  conditions:
+  - lastTransitionTime: "2020-09-15T18:15:29Z"
+    message: 'PipelineRun was paused.'
+    reason: PipelineRunPaused
+    status: "Unknown"
+    type: Succeeded
+```
+
+To "un-pause" a resource, the operator should set the `Status` field to empty,
+the Tekton controller will see the change and proceed with the Pipeline as
+normal. Upon unpausing, the `Succeeded` `Condition` `Reason` will be changed to
+`Running`:
+
+```
+status:
+  conditions:
+  - lastTransitionTime: "2020-09-15T19:46:34Z"
+    message: Not all Steps in the Task have finished executing
+    reason: Running
+    status: Unknown
+    type: Succeeded
+```
+
+### User Stories
+
+#### Platform Implementor
+
+As a platform implementor, I want to be able to control when a `PipelineRun` is
+able to run without introducing bespoke machinery.
+
+## Alternatives
+
+* Instead of adding this setting directly to the `PipelineRun` object,
```

Comment:
Will there be any problems if the client-go requeue operation is used here?

Reply:
Sorry, not sure I understand the question